frontend/src/utilities/hooks/use-identity.ts

```diff
@@ -0,0 +1,64 @@
+import IdentityRecord from "models/view-models/identity-record";
+import { useCallback } from "react";
+import UserLoginRecord from "models/view-models/user-login-record";
+import RoleService from "utilities/services/role-service";
+import UserService from "utilities/services/user-service";
+import UserRecord from "models/view-models/user-record";
+import RoleRecord from "models/view-models/role-record";
+
+/**
+ * Custom hook providing utility functions for the current identity of the user
+ */
+export default function useIdentity() {
+    const { get: getRoleApi } = RoleService.useGet();
+    const { get: getUserApi } = UserService.useGet();
+
+    /**
+     * Build the identity record including the current User and Role
+     *
+     * @param  {UserLoginRecord|undefined} (userLogin
+     * @returns Promise
+     */
+    const getHook = (userLogin: UserLoginRecord | undefined) => {
+        if (userLogin == null) {
+            return;
+        }
+
+        return getIdentity(userLogin);
+    };
+
+    const getIdentity = async (
+        userLogin: UserLoginRecord
+    ): Promise<IdentityRecord | undefined> => {
+        const role = await getRole(userLogin?.roleId);
+        const user = await getUser(userLogin?.userId);
+
+        if (user == null) {
+            return;
+        }
+
+        return new IdentityRecord({ role, user });
+    };
+
+    const getRole = async (
```

Comment:
Note: Was able to distill this code down to the point that getRole and getUser are near identical and we could abstract. Very tempted to keep going, but gotta keep trucking along.